### PR TITLE
Test that delete markers are not listed for version suspended buckets

### PR DESCRIPTION
**Pull request by KillianG**
Issue: CLDSRV-347

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js [hunk 1]
```diff
@@ -475,4 +475,56 @@ describe('Listing corner cases tests', () => {
             }
         });
     });
+
+    it('should not list DeleteMarkers for version suspended buckets', done => {
+        const obj = { name: 'testDeleteMarker.txt', value: 'foo' };
+        let objectCount = 0;
+        const Bucket = `versioning-bucket-test-delete-marker-${Date.now()}`;
```
Comment (jonathan-gramain) [inline tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js]: There's already a global called `Bucket`, it could be better to rename this new bucket to avoid confusion.

**Review by jonathan-gramain — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@zenko/cloudserver",
-  "version": "8.7.20",
+  "version": "8.7.21",
```
Comment (francoisferrand) [inline package.json]: 8.7.22 has been released already : please rebase & do the "expected" bump

(it is better to only add this kind of release bump at the last minute, once the PR has been approved)

File: tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js [hunk 3]
```diff
@@ -475,4 +475,56 @@ describe('Listing corner cases tests', () => {
...
+                    Key: obj.name,
+                    Body: obj.value,
+                }, err =>
+                next(err)),
+            next => s3.listObjectsV2({ Bucket: bucketName },
+                (err, res) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    objectCount = res.Contents.length;
```
Comment (francoisferrand) [inline tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js:506]: here we should check that we did indeed receive the object (testDeleteMarker.txt) : e.g. by going through the result and checking the key is found... (or that the result is "just" only that object)

File: tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js [hunk 4]
```diff
@@ -475,4 +475,56 @@ describe('Listing corner cases tests', () => {
...
+                    assert.strictEqual(
+                        headers['x-amz-delete-marker'], 'true');
+                    return next(err);
+                }),
+            next => s3.listObjectsV2({ Bucket: bucketName },
+                (err, res) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    assert.strictEqual(res.Contents.length, objectCount - 1);
```
Comment (francoisferrand) [inline tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js:524]: similarly, should have a stricter check here

File: package.json (hunk 2 above)
Comment (KillianG) [reply package.json]: Ready to re-review, will bump and resolve conflicts when PR is approved

**Review by welansari — APPROVED** [review]

**Review by francoisferrand — APPROVED** [review]
